app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsCorrectQtyCondition.php

```diff
@@ -177,6 +177,9 @@ private function isQuantityIncrementCheckFailed(
         StockItemConfigurationInterface $stockItemConfiguration,
         float $requestedQty
     ) : bool {
+        if ($stockItemConfiguration->isQtyDecimal() && !$stockItemConfiguration->isEnableQtyIncrements()) {
+            return false;
+        }
```

Comment:
Fix looks incorrect to me.
- why did we add isDecimal check to existing Increment validation? 

our desirable state to support decimal increment, so there should not be problem to have two independent checks

Reply:
@maghamed , I've tested simple and configurable product with decimal qty on this branch with fixes from magento-engcom/msi#1014 and magento-engcom/msi#1015 and looks like everything works fine . 
The only place I found, where decimal qty doesn't work is validator, mentioned above. And it's validation is pretty straightforward: if requested qty doesn't exactly divided by qty increments it will fail(Currently it doesn't considering qty decimal stock item configuration, as core magento does), So, if I put qty decimal validation into separate check, the overall validation will still fail.
Can you please advise, how to implement qty decimal validation into separate validator without failing overall validation?